domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs

```diff
@@ -3,67 +3,58 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using API.Configurations;
 using API.MeasurementsSyncer.Metrics;
 using API.MeasurementsSyncer.Persistence;
 using DataContext.Models;
 using DataContext.ValueObjects;
 using Measurements.V1;
 using Meteringpoint.V1;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace API.MeasurementsSyncer;
 
-public class MeasurementsSyncService
+public class MeasurementsSyncService(
+    ILogger<MeasurementsSyncService> logger,
+    ISlidingWindowState slidingWindowState,
+    Measurements.V1.Measurements.MeasurementsClient measurementsClient,
+    SlidingWindowService slidingWindowService,
+    IMeasurementSyncMetrics measurementSyncMetrics,
+    IMeasurementSyncPublisher measurementSyncPublisher,
+    Meteringpoint.V1.Meteringpoint.MeteringpointClient meteringPointsClient,
+    IOptions<MeasurementsSyncOptions> options)
 {
-    private readonly ISlidingWindowState _slidingWindowState;
-    private readonly Measurements.V1.Measurements.MeasurementsClient _measurementsClient;
-    private readonly SlidingWindowService _slidingWindowService;
-    private readonly IMeasurementSyncMetrics _measurementSyncMetrics;
-    private readonly IMeasurementSyncPublisher _measurementSyncPublisher;
-    private readonly Meteringpoint.V1.Meteringpoint.MeteringpointClient _meteringPointsClient;
-    private readonly ILogger<MeasurementsSyncService> _logger;
-
-    public MeasurementsSyncService(ILogger<MeasurementsSyncService> logger, ISlidingWindowState slidingWindowState,
-        Measurements.V1.Measurements.MeasurementsClient measurementsClient, SlidingWindowService slidingWindowService,
-        IMeasurementSyncMetrics measurementSyncMetrics, IMeasurementSyncPublisher measurementSyncPublisher, Meteringpoint.V1.Meteringpoint.MeteringpointClient meteringPointsClient)
-    {
-        _logger = logger;
-        _slidingWindowState = slidingWindowState;
-        _measurementsClient = measurementsClient;
-        _slidingWindowService = slidingWindowService;
-        _measurementSyncMetrics = measurementSyncMetrics;
-        _measurementSyncPublisher = measurementSyncPublisher;
-        _meteringPointsClient = meteringPointsClient;
-    }
+    private readonly MeasurementsSyncOptions _options = options.Value;
 
     public async Task FetchAndPublishMeasurements(MeteringPointSyncInfo syncInfo, MeteringPointTimeSeriesSlidingWindow slidingWindow,
         CancellationToken stoppingToken)
     {
         var synchronizationPoint = UnixTimestamp.Now().RoundToLatestHour();
         var fetchedMeasurements = await FetchMeasurements(slidingWindow, syncInfo.MeteringPointOwner, synchronizationPoint, stoppingToken);
```

Comment:
You are getting the measurements and then filtering them. Why not just get the ones you need? Also, why not only get 7 days old issuing contracts?